include/boost/json/detail/config.hpp

```diff
@@ -198,6 +198,16 @@
 # endif
 #endif
 
+#ifndef BOOST_JSON_UNREACHABLE
+# if defined(__GNUC__) || defined(__clang__)
+#  define BOOST_JSON_UNREACHABLE() __builtin_unreachable()
+# elif defined(_MSC_VER)
+#  define BOOST_JSON_UNREACHABLE() __assume(0)
+# else // infinite loops have undefined behavior
+#  define BOOST_JSON_UNREACHABLE() do { } while(1)
+# endif
+#endif
```

Comment:
@pdimov is this correct?

Reply:
The [[noreturn]] fallback is probably better than an infinite loop.